Lib/idlelib/stackviewer.py

```diff
@@ -133,19 +129,11 @@ def _stack_viewer(parent):  # htest #
     flist = PyShellFileList(top)
     try: # to obtain a traceback object
         intentional_name_error
-    except NameError:
-        exc_type, exc_value, exc_tb = sys.exc_info()
-    # inject stack trace to sys
-    sys.last_type = exc_type
-    sys.last_value = exc_value
-    sys.last_traceback = exc_tb
-
-    StackBrowser(top, flist=flist, top=top, tb=exc_tb)
-
-    # restore sys to original state
-    del sys.last_type
-    del sys.last_value
-    del sys.last_traceback
+    except NameError as e:
+        exc = e
+
+    StackBrowser(top, flist=flist, top=top, tb=exc)
```

Comment:
After stackbrowser definition changed above:
```suggestion
    stackbrowser(top, exc. flist=flist, top=top)
```
Test by running file.

Reply:
June 7: moved revised call into exception block.